integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -162,7 +165,10 @@ def _execute_sql(
             result = cursor.execute(sql_query, params)
         except Error as e:
             self._connection.rollback()
-            raise DocumentStoreError(error_msg) from e
+            sql_query_str = sql_query.as_string(cursor) if not isinstance(sql_query, str) else sql_query
+            detailed_error_msg = f"{error_msg}.\nSQL query: {sql_query_str} \nParameters: {params}"
+            raise DocumentStoreError(detailed_error_msg) from e
```

Comment:
unrelated to this PR, but should show useful info when the query fails

Reply:
Beware printing raw SQL queries, they can leak secrets and sensitive data.

Reply:
I've been thinking about it a bit. What do you suggest to show useful information without risk of leaks?

Reply:
We could print the SQL query in the debug logs and do something like:
```
detailed_error_msg = f"{error_msg}.\nYou can find the SQL query and the parameters in the debug logs"
```
It's not perfect but it should be ok for most of the cases